sitemap/sitemap-generator.ts

```diff
@@ -0,0 +1,130 @@
+const fs = require("fs");
+const convertor = require("xml-js");
+import type { ReactElement } from "react";
+import React from "react";
+
+const DEFAULT_CONFIG = {
+  lastmod: new Date().toISOString().slice(0, 10),
+  changefreq: "monthly",
+  priority: 0.8,
+};
+
+export type Config = {
+  lastmod?: string;
+  changefreq?: string;
+  priority?: number;
+};
+
+type rElement = ReactElement<any, any> | undefined;
+
+export class Generator {
+  _baseUrl: string;
+  _baseComponent: rElement[];
```

Comment:
i think this [lib](https://github.com/ekalinin/sitemap.js) could covered what u need to do, instead of rolling your own generator